zim/gui/widgets.py

```diff
@@ -1998,7 +1998,7 @@ def update_completion(self):
 
 		text = self.get_text()
 		if self._current_completion:
-			if text.startswith(self._current_completion) \
```

Comment:
No, this part of the code populates the list, but the filter function determines what is shown. So typing more text will constrain the results as the filter function selects matching entries. This optimization just means we don't bother to re-populate with a sub-set as the filter function already selects a subset.